gusto/domain.py

```diff
@@ -165,7 +165,7 @@ def __init__(self, mesh, dt, family, degree=None,
                 else:
                     metadata_value = None
                 for procid in range(1, comm_size):
-                    my_tag = comm_size*j + my_rank
+                    my_tag = comm_size*j + procid
```

Comment:
Thanks very much for spotting this!! I've checked the other times that we manually send/receive and I think this is the only one that was wrong.

Reply:
Should they be updated to gathers/scatters rather than manual send-receives? It feels like that is what is supposed to be happening here.

Reply:
Yes that almost certainly would be better. In the long run we should use pnetcdf and hopefully this should be handled for us ... I'm happy to follow up in a separate PR and change these

Reply:
Feel free to keep adding commits to this PR if you like, I don't think I'll be making any new changes.